piptools/utils.py

```diff
@@ -678,14 +678,34 @@ def parse_config_file(
             hint=f"Could not parse '{config_file !s}': {value_err !s}",
         )
 
-    # In a TOML file, we expect the config to be under `[tool.pip-tools]`
+    # In a TOML file, we expect the config to be under `[tool.pip-tools]`,
+    # `[tool.pip-compile]` or `[tool.pip-sync]`
     piptools_config: dict[str, Any] = config.get("tool", {}).get("pip-tools", {})
-    piptools_config = _normalize_keys_in_config(piptools_config)
-    piptools_config = _invert_negative_bool_options_in_config(
+    pipcompile_config: dict[str, Any] = (
+        config.get("tool", {}).get("pip-tools", {}).get("compile", {})
+    )
+    pipsync_config: dict[str, Any] = (
+        config.get("tool", {}).get("pip-tools", {}).get("sync", {})
+    )
+
+    config = piptools_config
+
+    if click_context.command.name == "pip-compile":
+        if pipcompile_config:
+            config.pop("compile")
+            config.update(pipcompile_config)
+    elif click_context.command.name == "pip-sync":
+        if pipsync_config:
+            config.pop("sync")
+            config.update(pipsync_config)
```

Comment:
This can probably be simplified further:
```suggestion
    cmd_to_section_map = {
        "pip-compile": "compile",
        "pip-sync": "sync",
    }
    config_section_name = cmd_to_section_map[click_context.command.name]
    piptools_config.update(piptools_config.pop(config_section_name, {}))
    piptools_config.pop("compile", {})
    piptools_config.pop("sync", {})
```
The above assumes we don't want those subsections in the object we'll be passing around...

Reply:
The map isn't really needed, though:
```suggestion
    config_section_name = click_context.command.name.removeprefix("pip-")
    piptools_config.update(piptools_config.pop(config_section_name, {}))
    piptools_config.pop("compile", {})
    piptools_config.pop("sync", {})
```

Reply:
This might be easier to grasp:
```suggestion
    config_section_name = click_context.command.name.removeprefix("pip-")
    piptools_config = {
        **piptools_config,
        "compile": {},
        "sync": {},
        **piptools_config.pop(config_section_name, {}),
    }

```